app/src/main/java/com/tromian/game/afproject/presentation/viewmodels/MoviesViewModel.kt

```diff
@@ -1,48 +1,43 @@
 package com.tromian.game.afproject.presentation.viewmodels
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
+import androidx.paging.cachedIn
+import com.tromian.game.afproject.domain.MovieListType
 import com.tromian.game.afproject.domain.models.Movie
 import com.tromian.game.afproject.domain.repository.MoviesRepository
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
 
 class MoviesViewModel(
-    private val repository: MoviesRepository
+    repository: MoviesRepository
 ) : ViewModel() {
 
-    private val _movieList = MutableLiveData<List<Movie>>()
-    val movieList: LiveData<List<Movie>> = _movieList
+    private var nowPlayingList = repository.getMovieListResultStream(MovieListType.NOW_PLAYING)
+        .cachedIn(viewModelScope)
```

Comment:
1) не смотрел глубоко доку и примеры по ViewPager. Точно в cachedIn должен быть ViewModelScope?
2) опять повторение кода 4 раза. ты можешь сделать свою extension fun типа:
```kotlin
fun Flow<PagingData<Movie>>.toStateFlow() = this.cachedIn(viewModelScope)
        .stateIn(viewModelScope, SharingStarted.Lazily, PagingData.empty())
```

Reply:
или еще вариант - сделать функцию
```
fun getMovieListStateFlow(param: MovieListType): StateFlow<PagingData<Movie>>
```

Reply:
1. В описании cachedIn (A common use case for this caching is to cache PagingData in a ViewModel. This can ensure that, upon configuration change (e.g. rotation),)

Reply:
2. Выбрал вариант 
```
fun Flow<PagingData<Movie>>.toStateFlow() = this.cachedIn(viewModelScope)
        .stateIn(viewModelScope, SharingStarted.Lazily, PagingData.empty())
```